tiledb/api/c_api_test_support/testsupport_capi_vfs.h

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file tiledb/api/c_api_test_support/testsupport_capi_vfs.h
```

Comment:
Lets rename this header as `c_api_test_support/testsupport_capi_vfs.h` is a bit redundant. `unit_capi_vfs.h` or `capi_ordinary_vfs.h` or something?

Also, what are your feelings on how often this wold be reused? Its certainly helpful cleaning up the VFS unit tests but I'm not sure if it'll be directly useful in other units? If you do think it'll be useful in other units then leave it here, but if you think its just for the `unit_capi_vfs.cc` then we should just move it along side that source file I think to communicate that its a helper for just that one test.

Reply:
@bekadavis9 Noted that this is mimicking the existing pattern so I'm gonna say ignore all this. I'd say its better to match what exists and then change things later if we decide on a different approach. I don't want to have that possible (probable?) bike shed derail this PR.